ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -17,6 +17,121 @@
 {-# LANGUAGE TupleSections              #-}
 {-# LANGUAGE TypeApplications           #-}
 
+{-|
+  = Snapshot managing and contents
+
+  Snapshotting a ledger state means saving a copy of the in-memory part of the
+  ledger state serialized as a file in the disk, as well as flushing differences
+  between the last snapshotted ledger state and the one that we are snapshotting
+  now onto the BackingStore. While the coupled implementation is in place, the
+  serialized file might also contain a serialized version of the in-memory UTxO
+  map carried by the legacy ledger states, as explained below.
+
+  == Startup
+
+  On startup, the node will:
+
+  1. Find the latest snapshot which will be a directory identified by the slot
+     number of the snapshot:
+
+        > <cardano-node data dir>
+        > ├── volatile
+        > ├── immutable
+        > └── ledger
+        >     ├── <slotNumber1>
+        >     │   ├── tables
+        >     │   └── state
+        >     ├── <slotNumber2>
+        >     │   ├── tables
+        >     │   └── state
+        >     └── <slotNumber3>
+        >         ├── tables
+        >         └── state
+
+  2. Depending on the snapshots found, there are two paths:
+
+       - If there is no snapshot to load, create a new @BackingStore@ with the
```

Comment:
If we use single quotes I think the haddock will reference to the `BackingStore` haddocks.